include/imex/Dialect/Dist/IR/DistOps.td

```diff
@@ -30,12 +30,41 @@ def Dist_Dialect : Dialect {
 
     // A longer description of our dialect.
     let description = [{
-            The dist dialect describes interfaces for interacting with
+        The dist dialect describes interfaces for interacting with
 	    a runtime which handles distributed aspects of PTensor operations.
         }];
 
+    let dependentDialects = [
+        "::imex::ptensor::PTensorDialect"
+    ];
+
     // The C++ namespace that the dialect class definition resides in.
     let cppNamespace = "::imex::dist";
+    let useDefaultTypePrinterParser = 1;
+}
+
+// common base classes for types in Dist dialect
+class Dist_Type<string name, string typeMnemonic, list<Trait> traits = []>
+    : TypeDef<Dist_Dialect, name, traits> {
+    let mnemonic = typeMnemonic;
+}
+
+def Dist_Info : Dist_Type<"DistInfo", "info">
+{
+  let summary = "A type for attaching distributed info information to a PTensor";
+  let description = [{
+    A distributed PTensor needs info information like offset and shape of local partition.
+    The DistInfo type is used to define operations to carry and extract such info data.
+  }];
+  let parameters = (ins "int32_t":$rank);
```

Comment:
Is `rank` similar to the ranks of process in a group like in MPI?

Reply:
In MLIR `rank` refers to the number of dimensions of a tensor. This is what is meant here.

You are right, we have a clash of terminology here. In later version I use `prank` for the process-rank.